pkg/startup/startup.go

```diff
@@ -956,10 +1025,22 @@ func checkConflictingNodes(ctx context.Context, client client.Interface, node *a
 
 // ensureDefaultConfig ensures all of the required default settings are
 // configured.
-func ensureDefaultConfig(ctx context.Context, cfg *apiconfig.CalicoAPIConfig, c client.Interface, node *api.Node) error {
+func ensureDefaultConfig(ctx context.Context, cfg *apiconfig.CalicoAPIConfig, c client.Interface, node *api.Node, clientset *kubernetes.Clientset) error {
 	// Ensure the ClusterInformation is populated.
 	// Get the ClusterType from ENV var. This is set from the manifest.
 	clusterType := os.Getenv("CLUSTER_TYPE")
+
+	if clientset != nil {
```

Comment:
Just continuing to mull on this, what do you think about actually querying the configmap only once higher up in the call stack, and then passing that information down to the appropriate functions instead of the configmap?

I think it has some tiny benefits, namely fewer API calls (each of which might fail), slightly less code duplication, and makes UTs easier (no need to pass a mock clientset or set up cluster access, can just pass a configmap struct or a boolean). 

WDYT?

Reply:
It's the trade off between fewer API calls in the `kubeadm` case but unnecessary calls otherwise. I was leaning towards the former but open to changing that.

Reply:
I don't think that's true - we always need to make one API call to query the configmap in order to determine if we're running on kubeadm.

In other words, right now we always query it twice. We could be querying it only once.